src/ReinforcementLearningCore/src/policies/q_based_policies/learners/approximators/neural_network_approximator.jl

```diff
@@ -101,22 +104,44 @@ function (model::GaussianNetwork)(rng::AbstractRNG, state; is_sampling::Bool=fal
         noise = Zygote.ignore() do
             send_to_device(device(model), randn(rng, Float32, size(μ)))
         end
-        z = μ .+ σ .* noise
+        z = model.normalizer.(μ .+ σ .* noise)
         if is_return_log_prob
             logp_π = sum(normlogpdf(μ, σ, z) .- (2.0f0 .* (log(2.0f0) .- z .- softplus.(-2.0f0 .* z))), dims = 1)
-            return tanh.(z), logp_π
+            return z, logp_π
         else
-            return tanh.(z)
+            return z
         end
     else
         return μ, logσ
     end
 end
 
+"""
+Sample `action_samples` actions from each state. Returns a 3D tensor with dimensions (action_size x action_samples x batch_size).
+`state` must be 3D tensor with dimensions (state_size x 1 x batch_size). Always returns the logpdf of each action along.
+"""
+function (model::GaussianNetwork)(rng::AbstractRNG, state, action_samples::Int)
+    x = model.pre(state)
+    μ, raw_logσ = model.μ(x), model.logσ(x)
+    logσ = clamp.(raw_logσ, log(model.min_σ), log(model.max_σ))
+    
+    σ = exp.(logσ)
+    noise = Zygote.ignore() do
+        send_to_device(device(model), randn(rng, Float32, (size(μ,1), action_samples, size(μ,3))...))
+    end
```

Comment:
You can simply pass a `RNG` on GPU to generate the noise directly to avoid the overhead of passing data from CPU to GPU.

Reply:
Yes indeed, here I copied the practice from the current implementation (see line 105 of the same file). But basically, since rng is passed by the algorithm and is decided by the user, we can just do 
```suggestion
    noise = Zygote.ignore() do
        randn(rng, Float32, (size(μ,1), action_samples, size(μ,3))...)
    end
```
and assume the user passed a rng on the same device as the GN?

Reply:
Yep